app/src/main/java/com/github/se/cyrcle/ui/list/List.kt

```diff
@@ -0,0 +1,449 @@
+
+                protectionMatch + rackTypeMatch + capacityMatch
+              }
+              .thenBy { (_, distance) -> distance })
+
+  Scaffold(
+      bottomBar = {
+        BottomBar(isListSelected = true, onNavigateToMap = { /* Navigate to Map */})
+      }) { innerPadding ->
+        LazyColumn(
+            modifier = Modifier.fillMaxSize().padding(innerPadding).padding(bottom = 16.dp),
+            contentPadding = PaddingValues(16.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp)) {
+              item {
+                FilterHeader(
+                    selectedProtection = selectedProtection,
+                    onProtectionSelected = { protection ->
+                      if (selectedProtection.contains(protection)) {
+                        selectedProtection = selectedProtection - protection
+                      } else {
+                        selectedProtection = selectedProtection + protection
+                      }
+                    },
+                    selectedRackTypes = selectedRackTypes,
+                    onRackTypeSelected = { rackType ->
+                      if (selectedRackTypes.contains(rackType)) {
+                        selectedRackTypes = selectedRackTypes - rackType
+                      } else {
+                        selectedRackTypes = selectedRackTypes + rackType
+                      }
+                    },
+                    selectedCapacities = selectedCapacities,
+                    onCapacitySelected = { capacity ->
+                      if (selectedCapacities.contains(capacity)) {
+                        selectedCapacities = selectedCapacities - capacity
+                      } else {
+                        selectedCapacities = selectedCapacities + capacity
+                      }
+                    })
+              }
+              items(sortedFilteredParkingSpots) { (parking, distance) ->
+                val matchedCriteria = mutableListOf<String>()
+
+                if (selectedProtection.isNotEmpty() &&
+                    selectedProtection.contains(parking.protection)) {
+                  matchedCriteria.add("Protection: ${parking.protection}")
+                }
+                if (selectedRackTypes.isNotEmpty() &&
+                    selectedRackTypes.contains(parking.rackType)) {
+                  matchedCriteria.add("Rack Type: ${parking.rackType}")
+                }
+                if (selectedCapacities.isNotEmpty() &&
+                    selectedCapacities.contains(parking.capacity)) {
+                  matchedCriteria.add("Capacity: ${parking.capacity}")
+                }
+
+                SpotCard(parking, distance, matchedCriteria)
+              }
+            }
+      }
+}
+
+// Extension fuction to toggle selection
+fun <T> Set<T>.toggle(item: T): Set<T> {
+  return if (this.contains(item)) this - item else this + item
+}
+
+fun calculateDistance(point1: Point, point2: Point): Double {
```

Comment:
We'll need to modify this to include viewModel-based distance modifications, but this works just fine for now